src/operator/mkldnn/mkldnn_memory-inl.h

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file mkldnn_memory-inl.h 
+ * \brief
+ * \author Chen, Xiaoming 
+*/
+
+#ifndef MXNET_OPERATOR_MKLDNN_MKLDNN_MEMORY_INL_H_
+#define MXNET_OPERATOR_MKLDNN_MKLDNN_MEMORY_INL_H_
+
+#include <dmlc/logging.h>
+#include "./mkldnn_cppwrapper.h"
+namespace mxnet {
+#if MXNET_USE_MKLDNN == 1
+template <typename DType>
+struct MKLMemoryDescriptorBase {
```

Comment:
This holds actual buffers.
MKLDNN's layer output results stores as different layout and memory size(I guess more strict memory alignment), which cannot be saved directly into ndarray/tblob.
We also do some first trial to fix this. After this PR, hope we can discuss about it. Thanks.